Project-Aurora/Project-Aurora/App.xaml.cs

```diff
@@ -124,7 +125,7 @@ protected override void OnStartup(StartupEventArgs e)
 
             Global.logger.Info("Loading Device Manager");
             Global.dev_manager.RegisterVariables();
-            Global.dev_manager.InitializeDevices();
+            Task.Run(async () => await Global.dev_manager.InitializeDevices());
```

Comment:
Maybe this could be reverted. Starting device updates on line 132 before they are initialized might make problems

Reply:
I had some issues with initialization when running this synchronously - I think that was really due to the YeeLight MusicMode issue (where initialization timed out after a while). I see what you mean though. I take a look tonight or tomorrow - thank you!

Reply:
It turns out I was using fire and forget because this is the same behavior that the Device Manager itself used to use.
```cs
        public void InitializeDevices()
        {
            if (suspended)
                return;

            var devices = DeviceContainers
                .Select(dc => dc.Device)
                .Where(device => !device.IsInitialized && !Global.Configuration.DevicesDisabled.Contains(device.GetType()));
            foreach (var device in devices)
            {
                device.InitializeTask = Task.Run(device.Initialize);
                device.InitializeTask.ContinueWith(task =>
                {
                    if (task.IsCompletedSuccessfully)
                    {
                        Global.logger.Info($"[Device][{device.DeviceName}] Initialized Successfully.");
                    }
                    else
                    {
                        Global.logger.Info($"[Device][{device.DeviceName}] Failed to initialize.");
                    }
                });
            }

            _InitializeOnceAllowed = false;
        }
```

That `Task.Run` inside the `foreach` starts all those tasks, but doesn't wait. 

I did find that some potential locking comes up because the app uses a lot of `async` methods from synchronous code, so I added `.ConfigureAwait(false)` to the `await` statements in the devices and the device manager.